backend/capellacollab/core/authentication/injectables.py

```diff
@@ -0,0 +1,112 @@
+# SPDX-FileCopyrightText: Copyright DB Netz AG and the capella-collab-manager contributors
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+import fastapi
+
+from capellacollab.core import database
+from capellacollab.core.authentication import jwt_bearer
+from capellacollab.projects import crud as projects_crud
+from capellacollab.projects.users import crud as projects_users_crud
+from capellacollab.projects.users import models as projects_users_models
+from capellacollab.users import crud as users_crud
+from capellacollab.users import models as users_models
+
+from . import helper
+
+
+class RoleVerification:
+    def __init__(self, required_role: users_models.Role, verify: bool = True):
+        self.required_role = required_role
+        self.verify = verify
+
+    def __call__(
+        self,
+        token=fastapi.Depends(jwt_bearer.JWTBearer()),
+        db=fastapi.Depends(database.get_db),
+    ) -> bool:
+        role = users_crud.get_user_by_name(
+            db=db, username=helper.get_username(token)
+        ).role
+        if (
+            role == users_models.Role.USER
```

Comment:
```suggestion
            role != users_models.Role.ADMIN
```

With this statement I do not have to know there are only 2 roles in the system.

Reply:
Well spotted. I think this is sufficient for now, as it at least prevents the case where we add another user role that would then have come through this check.